### PR TITLE
Create edge diagnostics adapter connector

### PR DESCRIPTION
**Pull request by sarvaje**

### CODE REVIEW

File: src/lib/connectors/connector-edge15/connector-edge15-launcher.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+                        return;
+                    });
+            };
+
+            poll();
+        });
+    }
+
+    private isWin10() {
```
Comment (molant) [inline src/lib/connectors/connector-edge15/connector-edge15-launcher.ts]: Missing function descriptions in the following ones.

File: src/lib/connectors/connector-edge15/connector-edge15.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/**
+ * @fileoverview Collector that uses the Chrome Debugging protocol to
+ * load a site and do the traversing.
```
Comment (molant) [inline src/lib/connectors/connector-edge15/connector-edge15.ts]: > Connector for Edge 15 that [edge-diagnostics-adapter](https://github.com/Microsoft/Edge-diagnostics-adapter) to load a site and do the traversing.

File: src/lib/index.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/**
+ * @fileoverview Edge Diagnostics Adapter connector for Sonar
```
Comment (molant) [inline src/lib/index.ts]: lower case `sonar`

File: src/lib/connectors/connector-edge15/connector-edge15-launcher.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+    /** Removes all references to the client used by `isDebuggerReady`. */
+    private cleanup(client: net.Socket) {
+        client.removeAllListeners();
+        client.end();
+        client.destroy();
+        client.unref();
+    }
+
+    /** Checks if the debugger is ready by trying to connect to port `9222`. */
```
Comment (sarvaje) [inline src/lib/connectors/connector-edge15/connector-edge15-launcher.ts]: Yep

**Comment by sarvaje** [comment]
@molant done!

File: package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
+  "dependencies": {
+    "edge-diagnostics-adapter": "file:../edge-diagnostics-adapter"
```
Comment (alrra) [inline package.json]: @sarvaje Is there any reason why this is a local dependency?

Comment (alrra) [inline package.json]: See #6.

Comment (molant) [inline package.json]: Yes, see https://github.com/sonarwhal/edge15-connector/issues/3

**Pull request merged**